<a href="https://colab.research.google.com/github/RitukumariBE/Numpy-Basics/blob/main/Numpy_Student_Marks_Analytics.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Numpy student Marks Analytics
- In this project we are going to work on a dataset which in telling about the marks of student and we are going to analysis it and taking output from it
-Tools used - NumPy

## Dataset Creation

In [1]:
import numpy as np


### This dataset includes
- Their is 10 rows and the rows corresponds to the students.
- Their is 5 columns and columns corresponds to the subjects.

In [2]:
# This array has 10 rows (students) and 5 columns (subjects)
marks = np.random.randint(1,100,50).reshape(10,5)

In [3]:
marks.shape

(10, 5)

In [4]:
marks

array([[80, 25, 37, 35, 63],
       [85, 17, 56, 57, 67],
       [55, 36, 99, 40,  9],
       [58, 62, 19, 73, 73],
       [54, 44, 54, 83, 48],
       [55, 83, 18, 80,  1],
       [90, 10, 95, 87, 44],
       [27, 69, 88, 56,  7],
       [54, 91,  4, 39, 96],
       [93, 10,  3, 54, 50]])

## Understanding the Data


- To show the marks of the 1st student

In [5]:
marks[0] # accessing the marks of student 1

array([80, 25, 37, 35, 63])

- To check the marks of all students in the 2nd subject

In [6]:
marks[:,1] # accessing the subject 2 marks of all students

array([25, 17, 36, 62, 44, 83, 10, 69, 91, 10])

- To check the last two subject marks of first three students

In [7]:
marks[0:3,3:]

array([[35, 63],
       [57, 67],
       [40,  9]])

## Subject-wise Analysis
- In this section we are going to analysis subject wise performance using the statistical measures

### Average of each subject

In [8]:
average1 = marks.mean(axis=0)
average1

array([65.1, 44.7, 47.3, 60.4, 45.8])

In [9]:
average1.max() # the hightest average among all the subjects

np.float64(65.1)

### Maximum of each subject

In [10]:
marks.max(axis=0)

array([93, 91, 99, 87, 96])

### Minimum of each subjecs

In [11]:
marks.min(axis = 0)

array([27, 10,  3, 35,  1])


- Highest average comes in the subject 1 with 60.4 . The 5th subject seems most difficult as the average comes out 27.3 . The performance is not looking balanced as one id=s 60 where other is 27 .

## Student-Wise Analysis

In [18]:
avg = marks.sum(axis=1)
avg

array([240, 282, 239, 285, 283, 237, 326, 247, 284, 210])

In [17]:
avg / 5

1.0

## Student Ranking

## Pass-Fail Classification

## Key Insights & Conclusion